### PR TITLE
Set edition in use_testthat()

### LINKED ISSUE #1185: Automatically use testthat 3e
When creating a package and you have testthat 3.0.0 or later installed.

We may also want to provide `use_testthat_3e()` function.

### PR DESCRIPTION
**Pull request by hadley**
Fixes #1185

### CODE REVIEW

File: R/test.R [hunk 1]
```diff
@@ -40,6 +45,22 @@ use_testthat_impl <- function() {
...
+  testthat_version <- packageVersion("testthat")[[1, 1]]
+  if (is.null(edition)) {
+    testthat_version
+  } else {
+    if (!is.numeric(edition) || length(edition) != 1) {
+      ui_stop("`edition` must be a single number")
+    }
+    if (edition > testthat_version) {
+      vers <- packageVersion("testthat")
+      ui_stop("`edition` ({edition}) not available in installed testthat ({vers}")
```
Comment (jennybc) [inline R/test.R:58]:
```suggestion
      ui_stop("`edition` ({edition}) not available in installed testthat ({vers})")
```

File: tests/testthat/test-test.R [hunk 2]
```diff
@@ -0,0 +1,9 @@
+test_that("check_edition() validates inputs", {
+  expect_error(check_edition(20), "not available")
+  expect_error(check_edition("x"), "single number")
+  expect_equal(check_edition(1.5), 1)
+
+  expect_equal(check_edition(), packageVersion("testthat")[[1, 1]])
+  expect_equal(2 * 2, 4)
```
Comment (jennybc) [inline tests/testthat/test-test.R:7]: What's this about?

**Review by jennybc — APPROVED** [review]

File: R/test.R [hunk 3]
```diff
@@ -13,23 +15,26 @@
...
   ui_todo(
     "Call {ui_code('use_test()')} to initialize a basic test file and open it \\
     for editing."
   )
 }
 
-use_testthat_impl <- function() {
+use_testthat_impl <- function(edition = NULL) {
   check_installed("testthat")
   if (utils::packageVersion("testthat") < "2.1.0") {
```
[the remark this answers is not in the document]
Comment (hadley) [reply R/test.R:29]: You mean encourage them to call this function for existing packages?

File: tests/testthat/test-test.R (hunk 2 above)
Comment (hadley) [reply tests/testthat/test-test.R:7]: Oops, accidentally forgot to delete part of the tempalte

File: R/test.R (hunk 3 above)
[the remark this answers is not in the document]
Comment (jennybc) [reply R/test.R:29]: Not exactly. I meant if we find that someone's starting a new package, setting up testthat, and they don't have the current CRAN version of testthat, we encourage them to update testthat first.

[the remark this answers is not in the document]
Comment (hadley) [reply R/test.R:29]: Oh hmmmmmmmmmmmmmmmmmmmmm

**Pull request merged**